### ISSUE TITLE
(module name): short issue description

### ISSUE THREAD

**Issue by markussiebert** [aws/aws-cdk#17684]
### Description

RDS Instances (maybe Clusters, didn't checked that) should allow adding Securitygroups

### Use Case

We have predefined SGs we want to add to all Resources.

### Proposed Solution

RDS Instances should stick to the implementation of EC2 Instances

https://github.com/aws/aws-cdk/blob/dd5e12d43231d991971de3121928377569b2dd63/packages/%40aws-cdk/aws-ec2/lib/instance.ts#L430

https://github.com/aws/aws-cdk/blob/dd5e12d43231d991971de3121928377569b2dd63/packages/%40aws-cdk/aws-ec2/lib/instance.ts#L303

instead of holding the securitygroups only inside the constructor.

### Other information

no

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by skinny85** [comment]
Thanks for opening the issue @markussiebert. Actually, I think the problem is [here](https://github.com/aws/aws-cdk/blob/ec4187c26d68df970d72d0e766d7d27b42e8b784/packages/%40aws-cdk/aws-ec2/lib/instance.ts#L320-L321) . Instead of using `this.securityGroups` [here](https://github.com/aws/aws-cdk/blob/ec4187c26d68df970d72d0e766d7d27b42e8b784/packages/%40aws-cdk/aws-ec2/lib/instance.ts#L337), we should use `this.connections.securityGroups` instead. That would allow you go go `instance.connections.addSecurityGroup()`.

**Comment by markussiebert** [comment]
Good point @skinny85 - first I thought the same! But, now I think the way it's implement for EC2 Instances is actually really good.

While thinking about this I found:
https://github.com/aws/aws-cdk-rfcs/issues/51
really interesting

If I read this:
```
/**
 * The goal of this module is to make possible to write statements like this:
 *
 *  ```ts
 *  database.connections.allowFrom(fleet);
 *  fleet.connections.allowTo(database);
 *  rdgw.connections.allowFromCidrIp('0.3.1.5/86');
 *  rgdw.connections.allowTrafficTo(fleet, new AllPorts());
 *  ```
 *
 * The insight here is that some connecting peers have information on what ports should
 * be involved in the connection, and some don't.
 */
```
I personally don't want to have all Securitygroups in the connections. If I write a statement database.connections.allowFrom(myec2Instance) I don't want my Securitygroup granting access for administrators extended the same way. 

That's how the ec2 way works.

But I would like to see all L2/L3 Constructs work the same way :-) (some Constructs say addSecurityGroup, others say addSecurityGroups ... some won't allow this...)

**Comment by skinny85** [comment]
I don't understand what's the final proposal from your post, but fell free to open a PR 😉.

**Comment by markussiebert** [comment]
Sorry - at the moment I have no proposal. I would like that all resources behave in the same way. So I think I have to look into how to contribute to the rfcs ... 

Because every cdk resource behaves in a different way I was not able to add my SecurityGroup to a lot resources apart from some ec2 resources and now using another approach with prefix lists and an aspect, that creates an ingress for each security group found in the cdk app.
